capstone/helpers/image_processing.py

```diff
@@ -0,0 +1,111 @@
+import matplotlib.pyplot as plt
+import numpy as np
+import skimage.io as io
+import skimage.transform as transform
+import skimage.exposure as exposure
+import glob
+import os
+
+
+def print_columns(data, cols):
+    """
+    Splits a list into sublists and prints it out in columns
+    :params data: list of data to be printed
+    :params cols: number of columns
+    :return: print the data and return none
+    """
+    num_per_col = int(np.ceil(len(data) / cols))
+    split = [data[i:i + num_per_col] for i in range(0, len(data), num_per_col)]
+    for row in zip(*split):
+        print("".join(str.ljust(i, 30) for i in row))
+
+
+def standardize(image_name, size, path):
+    """
+    Standardizes an image to a particular size and rotation
+    :params image_name: name of new image
+    :params size: standard image size
+    :params path: path to images directory
+    :return: standardized image as numpy array
+    """
+    image = io.imread(path + image_name)
+    existing_shape = image.shape
+    if existing_shape[1] > existing_shape[0]:
+        image = transform.rotate(image, 90, resize=True)
+    if image.shape != size:
+        image = transform.resize(image, size)
+    return image
+
+
+def mean_image(image_names, shape, path):
+    """
+    Calculates the average image from a list of image names
+    :params image_names: list of image names
+    :params shape: standard image shape
+    :params path: path to images directory
+    :return: average image as numpy array
+    """
+    N = len(image_names)
+    avg_image = np.zeros(shape)
+    for name in image_names:
+        image = standardize(name, shape, path)
+        avg_image += image / N
+    return avg_image
+
+
+def get_image_names(name):
+    base = '/Users/simonbedford/Documents/Coding/datascience/capstone/data/food-101/images/'
+    os.chdir(base + name)
```

Comment:
`chdir` is not that usual. Better would probably be to call on line 60 `glob.glob` on full path, eg.

```
path = os.path.join('/Users/simonbedford/Documents/Coding/datascience/capstone/data/food-101/images/', name)
images = []
for file in glob.glob(os.path.join(path, "*.jpg")):
    ...
```